LaunchDarkly/LaunchDarkly/ServiceObjects/Cache/LDFileCache.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import OSLog
+
+public final class LDFileCache: KeyedValueCaching {
```

Comment:
An additional consideration with the debouncer is there will be a change in the when data becomes durable, where that happens, and the probability of it happening.

With UserDefaults, it is my understanding that the API guarantees eventual durability except in the case of OS/phone battery failure.  The issue you linked to in the PR description where flicker is seen on relaunch is a result of the OS not sharing the view of in memory caching of UserDefaults across processes.  Definitely an issue, but not trivial to address without blocking logic somewhere.

With the debouncer, there is a window of time where if two updates come in barely before the application is killed, the second update will affect flag evaluations for a moment, but then may not make it to persistence.  This could have the same effect of "flicker" where the next app launch initializes with flag data from "the past", similar to the UserDefaults case.

Based on @keelerm84's comment above, if the debouncer goes with this file, this behavior would be in your control.